docs/discovery.md

```diff
@@ -0,0 +1,274 @@
+docker -H tcp://<swarm_ip:swarm_port> ps
+docker -H tcp://<swarm_ip:swarm_port> logs ...
+...
+```
+
+You can list the nodes in your cluster.
+
+```bash
+swarm list etcd://<etcd_ip>/<path>
+<node_ip:2375>
+```
+
+### Using consul
+
+On each of your nodes, start the Swarm agent. The node IP address
+doesn't need to be public as long as the Swarm manager can access it.
+
+```bash
+swarm join --addr=<node_ip:2375> consul://<consul_addr>/<path>
+```
+
+Start the manager on any machine or your laptop.
+
+```bash
+swarm manage -H tcp://<swarm_ip:swarm_port> consul://<consul_addr>/<path>
+```
+
+And then use the regular Docker commands.
+
+```bash
+docker -H tcp://<swarm_ip:swarm_port> info
+docker -H tcp://<swarm_ip:swarm_port> run ...
+docker -H tcp://<swarm_ip:swarm_port> ps
+docker -H tcp://<swarm_ip:swarm_port> logs ...
+...
+```
+
+You can list the nodes in your cluster.
+
+```bash
+swarm list consul://<consul_addr>/<path>
+<node_ip:2375>
+```
+
+### Using zookeeper
+
+On each of your nodes, start the Swarm agent. The node IP doesn't have
+to be public as long as the swarm manager can access it.
+
+```bash
+swarm join --addr=<node_ip:2375> zk://<zookeeper_addr1>,<zookeeper_addr2>/<path>
+```
+
+Start the manager on any machine or your laptop.
+
+```bash
+swarm manage -H tcp://<swarm_ip:swarm_port> zk://<zookeeper_addr1>,<zookeeper_addr2>/<path>
+```
+
+You can then use the regular Docker commands.
+
+```bash
+docker -H tcp://<swarm_ip:swarm_port> info
+docker -H tcp://<swarm_ip:swarm_port> run ...
+docker -H tcp://<swarm_ip:swarm_port> ps
+docker -H tcp://<swarm_ip:swarm_port> logs ...
+...
+```
+
+You can list the nodes in the cluster.
+
+```bash
+swarm list zk://<zookeeper_addr1>,<zookeeper_addr2>/<path>
+<node_ip:2375>
+```
+
+### Using a static list of IP addresses
+
+Start the manager on any machine or your laptop
+
+```bash
+swarm manage -H <swarm_ip:swarm_port> nodes://<node_ip1:2375>,<node_ip2:2375>
+```
+
+Or
+
+```bash
+swarm manage -H <swarm_ip:swarm_port> <node_ip1:2375>,<node_ip2:2375>
+```
+
+Then use the regular Docker commands.
+
+```bash
+docker -H <swarm_ip:swarm_port> info
+docker -H <swarm_ip:swarm_port> run ...
+docker -H <swarm_ip:swarm_port> ps
+docker -H <swarm_ip:swarm_port> logs ...
+...
+```
+
+### Range pattern for IP addresses
+
+The `file` and `nodes` discoveries support a range pattern to specify IP
+addresses, i.e., `10.0.0.[10:200]` will be a list of nodes starting from
+`10.0.0.10` to `10.0.0.200`.
+
+For example for the `file` discovery method.
+
+```bash
+$ echo "10.0.0.[11:100]:2375"   >> /tmp/my_cluster
+$ echo "10.0.1.[15:20]:2375"    >> /tmp/my_cluster
+$ echo "192.168.1.2:[2:20]375"  >> /tmp/my_cluster
+```
+
+Then start the manager.
+
+```bash
+swarm manage -H tcp://<swarm_ip:swarm_port> file:///tmp/my_cluster
+```
+
+And for the `nodes` discovery method.
+
+```bash
+swarm manage -H <swarm_ip:swarm_port> "nodes://10.0.0.[10:200]:2375,10.0.1.[2:250]:2375"
+```
+
+## Contributing a new discovery backend
```

Comment:
Sounds good to me.  I'll make this fix and push.  I can't merge on your project --- if you want to give me access, I promise to be really careful.